cmd/tether/attach_linux.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"net"
+	_ "net/http/pprof"
+	"os"
+	"syscall"
+	"time"
+	"unsafe"
+
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/terminal"
+	"golang.org/x/net/context"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/vmware/vic/lib/portlayer/attach"
+	"github.com/vmware/vic/pkg/serial"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+var backchannelMode = os.ModePerm
+
+func backchannel(ctx context.Context, conn *net.Conn) error {
+	defer trace.End(trace.Begin("establish tether backchannel"))
+
+	// HACK: currently RawConn dosn't implement timeout so throttle the spinning
```

Comment:
I still don't understand this comment and I think I've read this code more than a dozen times.  I think it's safe to nuke it.